toolchain/base/canonical_value_store.h

```diff
@@ -17,18 +17,18 @@ namespace Carbon {
 // A wrapper for accumulating immutable values with deduplication, providing IDs
 // to later retrieve the value.
 //
-// `IdT::ValueType` must represent the type being indexed.
+// `ValueT` must represent the type being indexed.
```

Comment:
Since you're here:
```suggestion
// `ValueT` represents the type being stored.
```
(In `x[y]`, I'd interpret "the type being indexed" as the type of `x`, not the type of `y` or `x[y]`.)